src/elm/Pages/Build/Logs.elm

```diff
@@ -158,6 +158,69 @@ getLog resource get logs =
         |> List.head
 
 
+{-| addLog : takes incoming log and logs and adds log when not present
+-}
+addLog : Log -> Logs -> Logs
+addLog incomingLog logs =
+    RemoteData.succeed
+        (safeDecodeLogData incomingLog)
+        :: logs
+
+
+{-| updateLog : takes incoming log and logs and updates the appropriate log data
+-}
+updateLog : Log -> Logs -> Logs
+updateLog incomingLog logs =
+    updateIf
+        (\log ->
+            case log of
+                RemoteData.Success log_ ->
+                    incomingLog.id == log_.id && incomingLog.rawData /= log_.rawData
+
+                _ ->
+                    True
+        )
+        (\_ ->
+            RemoteData.succeed <|
+                safeDecodeLogData incomingLog
+        )
+        logs
+
+
+{-| safeDecodeLogData : takes log and decodes the data if it exists and does not exceed the size limit.
+-}
+safeDecodeLogData : Log -> Log
+safeDecodeLogData log =
+    { log
+        | decodedLogs =
+            if isEmpty log then
+                logEmptyMessage
+
+            else if exceedsSizeLimit log then
+                logSizeExceededMessage
+
+            else
+                Util.base64Decode log.rawData
```

Comment:
moving the `Util.base64Decode` call behind the size limit has improved the UI performance drastically. 

base64 decoding accounts for the majority of page-unresponsiveness when viewing logs